### PR TITLE
Make use of the backup mode passed in protobuf message

### PR DESCRIPTION
**Pull request by MattFellows**
This is an attempt at solving
https://github.com/thelastpickle/cassandra-medusa/issues/389

Criticisms and comments gladly recieved...

### CODE REVIEW

**Review by jsanda — APPROVED** [review]
Thanks for the PR! Sorry it took so long for the review.

File: medusa/service/grpc/server.py [hunk 1]
```diff
@@ -44,11 +46,14 @@ def __init__(self, config):
         self.storage = Storage(config=self.config.storage)
 
     def Backup(self, request, context):
-        logging.info("Performing backup {}".format(request.name))
+        logging.info("Performing backup {} (type={})".format(request.name, request.mode))
         resp = medusa_pb2.BackupResponse()
-        # TODO pass the staggered and mode args
+        # TODO pass the staggered arg
+        mode = BACKUP_MODE_DIFFERENTIAL
+        if medusa_pb2.Mode.FULL == request.mode:
```
Comment (jsanda) [inline medusa/service/grpc/server.py:53]: @MattFellows I missed this on my initial review. `medusa_pb2.Mode.FULL` is wrong. It should be `medusa_pb2.BackupRequest.Mode.FULL`. It is causing some integration tests to fail.

**Review by jsanda — CHANGES_REQUESTED** [review]

File: medusa/service/grpc/server.py (hunk 1 above)
Comment (MattFellows) [reply medusa/service/grpc/server.py]: There are 3 ways of referencing that enum that seem to pass local testing - you could of just told me that the 3rd was was wrong  (always nice to get a Monty Hall Problem reference in ;) )

**Comment by jsanda** [comment]
@emerkle826 can you take a look at these two integration test failures? I am not sure why they are failing.

**Comment by emerkle826** [comment]
The tests that are failing are ones that run against Cassandra trunk. The tests pass against 4.0.0 GA, so it's something in trunk that's been recently added. I'll dig in to see what's going on and figure out what needs to be changed.

**Comment by emerkle826** [comment]
The test failure is due to a non-backward compatible change in some internal Cassandra APIs that Management API uses. It's not an issue with Medusa, so I think this PR can be merged and ignore the test failures for now. The fix will have to be in the Management API project [here](https://github.com/k8ssandra/management-api-for-apache-cassandra), and I'll create a ticket over there to address it.

**Comment by emerkle826** [comment]
For the record, the Management API issue is being tracked [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/122)

**Comment by jsanda** [comment]
@emerkle826 thank you for following up this. Do you think we are good to merge since the failures are unrelated to the changes in the PR?

**Review by emerkle826 — APPROVED** [review]

**Pull request merged**